causalpy/expt_diff_in_diff.py

```diff
@@ -0,0 +1,203 @@
+            .groupby(self.time_variable_name)
+            .first()
+            .reset_index()
+        )
+        assert not self.x_pred_control.empty
+        (new_x,) = build_design_matrices([self._x_design_info], self.x_pred_control)
+        self.y_pred_control = self.model.predict(np.asarray(new_x))
+
+        # predicted outcome for treatment group
+        self.x_pred_treatment = (
+            self.data
+            # just the treated group
+            .query(f"{self.group_variable_name} == 1")
+            # drop the outcome variable
+            .drop(self.outcome_variable_name, axis=1)
+            # We may have multiple units per time point, we only want one time point
+            .groupby(self.time_variable_name)
+            .first()
+            .reset_index()
+        )
+        assert not self.x_pred_treatment.empty
+        (new_x,) = build_design_matrices([self._x_design_info], self.x_pred_treatment)
+        self.y_pred_treatment = self.model.predict(np.asarray(new_x))
+
+        # predicted outcome for counterfactual. This is given by removing the influence
+        # of the interaction term between the group and the post_treatment variable
+        self.x_pred_counterfactual = (
+            self.data
+            # just the treated group
+            .query(f"{self.group_variable_name} == 1")
+            # just the treatment period(s)
+            .query("post_treatment == True")
+            # drop the outcome variable
+            .drop(self.outcome_variable_name, axis=1)
+            # We may have multiple units per time point, we only want one time point
+            .groupby(self.time_variable_name)
+            .first()
+            .reset_index()
+        )
+        assert not self.x_pred_counterfactual.empty
+        (new_x,) = build_design_matrices(
+            [self._x_design_info], self.x_pred_counterfactual, return_type="dataframe"
+        )
+        # INTERVENTION: set the interaction term between the group and the
+        # post_treatment variable to zero. This is the counterfactual.
+        for i, label in enumerate(self.labels):
+            if "post_treatment" in label and self.group_variable_name in label:
+                new_x.iloc[:, i] = 0
+        self.y_pred_counterfactual = self.model.predict(np.asarray(new_x))
+
+        # calculate causal impact
+        if isinstance(self.model, PyMCModel):
+            # This is the coefficient on the interaction term
+            coeff_names = self.model.idata.posterior.coords["coeffs"].data
+            for i, label in enumerate(coeff_names):
+                if "post_treatment" in label and self.group_variable_name in label:
+                    self.causal_impact = self.model.idata.posterior["beta"].isel(
+                        {"coeffs": i}
+                    )
+        elif isinstance(self.model, ScikitLearnModel):
+            # This is the coefficient on the interaction term
+            # TODO: THIS IS NOT YET CORRECT ?????
```

Comment:
Is there an issue for this TODO?

Reply:
Issue created https://github.com/pymc-labs/CausalPy/issues/390